test/unit/org/apache/cassandra/io/util/ThreadLocalReadAheadBufferTest.java

```diff
@@ -139,10 +139,9 @@ private Gen<InputData> lastBlockReads()
 
     private Gen<InputData> randomReads()
     {
-        int blockSize = new DataStorageSpec.IntKibibytesBound("256KiB").toBytes();
         return arbitrary().pick(List.of(files))
                          .flatMap((file) ->
-                                  lists().of(longs().between(fileSize(file) - blockSize, fileSize(file)).zip(integers().between(1, 100), Pair::create))
+                                  lists().of(longs().between(0, fileSize(file)).zip(integers().between(1, 100), Pair::create))
```

Comment:
do we test only a single block read by using small sizes in 1..100 range ? I is a valid scenario to read more than 1 block?

Reply:
we disallow that case